catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardBackupFileStorage.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Override
+    public String getDescription() {
+        return (String) describableProperties.get(DESCRIPTION);
+    }
+
+    @Override
+    public String getOrganization() {
+        return (String) describableProperties.get(ORGANIZATION);
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void deleteData(String id) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException(
+                    "Unable to delete stored data; no output directory specified.");
+        }
+
+        deleteBackupIfPresent(id);
+    }
+
+    public void storeData(String id, byte[] data) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException("Unable to store data; no output directory specified.");
+        }
+
+        if (data == null) {
+            throw new MetacardBackupException("No data to store");
+        }
+
+        Path metacardPath = getMetacardDirectory(id);
+        if (metacardPath == null) {
+            LOGGER.debug(String.format("Unable to create metacard path directory for %s", id));
+            throw new MetacardBackupException("Unable to create metacard backup directory");
+        }
+
+        try {
+            Path parent = metacardPath.getParent();
+            if (parent != null) {
+                Files.createDirectories(parent);
+            }
+            Files.createFile(metacardPath);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to create backup file {}.  File may already exist.",
+                    metacardPath,
+                    e);
+        }
+
+        try (OutputStream outputStream = new FileOutputStream(metacardPath.toFile())) {
+            IOUtils.write(data, outputStream);
+        }
+    }
+
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
+    public String getOutputDirectory() {
+        return outputDirectory;
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        Object outputDirectory = properties.get(OUTPUT_DIRECTORY_PROPERTY);
+        if (outputDirectory instanceof String && StringUtils.isNotBlank((String) outputDirectory)) {
+            this.outputDirectory = (String) outputDirectory;
+            LOGGER.debug("Updating {} with {}", OUTPUT_DIRECTORY_PROPERTY, outputDirectory);
+        }
+
+        Object id = properties.get(ID);
+        if (id instanceof String) {
+            setId((String) id);
+        }
+    }
+
+    private void deleteBackupIfPresent(String filename) throws MetacardBackupException {
+        Path metacardPath = getMetacardDirectory(filename);
+        if (metacardPath == null) {
+            LOGGER.trace(String.format("Unable to delete backup for  %s", filename));
+            throw new MetacardBackupException("Unable to delete backup");
```

Comment:
Might be worthwhile to use the same string that you use in the trace statement here (includes id).  Otherwise you could drop `String.format` in the above line